### ISSUE TITLE
What can't be installed?

### ISSUE THREAD

**Issue by 1596944197** [bastengao/react-native-serial-port-api#25]
When I install react native serial port API through yarn, I cannot install the software by running yarn Android. The following is the error message
![image](https://user-images.githubusercontent.com/68650762/163776209-4e89b95f-f2e4-4f71-a973-8dff98691f23.png)
```
> Task :app:processRUIJIA_WIFI_RELEASEDebugManifest FAILED

See http://g.co/androidstudio/manifest-merger for more information about the manifest merger.


Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.0.1/userguide/command_line_interface.html#sec:command_line_warnings
733 actionable tasks: 203 executed, 526 from cache, 4 up-to-date
watchman warning:  Recrawled this watch 1 times, most recently because:
MustScanSubDirs UserDroppedTo resolve, please review the information on
https://facebook.github.io/watchman/docs/troubleshooting.html#recrawl
To clear this warning, run:
`watchman watch-del '/Users/smartop/Desktop/code/smartop-hospital-wifi' ; watchman watch-project '/Users/smartop/Desktop/code/smartop-hospital-wifi'`

Recrawled this watch 1 times, most recently because:
MustScanSubDirs UserDroppedTo resolve, please review the information on
https://facebook.github.io/watchman/docs/troubleshooting.html#recrawl
To clear this warning, run:
`watchman watch-del '/Users/smartop/Desktop/code/smartop-hospital-wifi' ; watchman watch-project '/Users/smartop/Desktop/code/smartop-hospital-wifi'`

/Users/smartop/Desktop/code/smartop-hospital-wifi/android/app/src/debug/AndroidManifest.xml:23:7-33 Error:
        Attribute application@allowBackup value=(true) from AndroidManifest.xml:23:7-33
        is also present at [:react-native-serial-port-api] AndroidManifest.xml:13:9-36 value=(false).
        Suggestion: add 'tools:replace="android:allowBackup"' to <application> element at AndroidManifest.xml:7:5-117 to override.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processRUIJIA_WIFI_RELEASEDebugManifest'.
> Manifest merger failed : Attribute application@allowBackup value=(true) from AndroidManifest.xml:23:7-33
        is also present at [:react-native-serial-port-api] AndroidManifest.xml:13:9-36 value=(false).
        Suggestion: add 'tools:replace="android:allowBackup"' to <application> element at AndroidManifest.xml:7:5-117 to override.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 55s

error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup. Run CLI with --verbose flag for more details.
Error: Command failed: ./gradlew app:installRUIJIA_WIFI_RELEASEDebug -PreactNativeDevServerPort=8081
watchman warning:  Recrawled this watch 1 times, most recently because:
MustScanSubDirs UserDroppedTo resolve, please review the information on
https://facebook.github.io/watchman/docs/troubleshooting.html#recrawl
To clear this warning, run:
`watchman watch-del '/Users/smartop/Desktop/code/smartop-hospital-wifi' ; watchman watch-project '/Users/smartop/Desktop/code/smartop-hospital-wifi'`

Recrawled this watch 1 times, most recently because:
MustScanSubDirs UserDroppedTo resolve, please review the information on
https://facebook.github.io/watchman/docs/troubleshooting.html#recrawl
To clear this warning, run:
`watchman watch-del '/Users/smartop/Desktop/code/smartop-hospital-wifi' ; watchman watch-project '/Users/smartop/Desktop/code/smartop-hospital-wifi'`

/Users/smartop/Desktop/code/smartop-hospital-wifi/android/app/src/debug/AndroidManifest.xml:23:7-33 Error:
        Attribute application@allowBackup value=(true) from AndroidManifest.xml:23:7-33
        is also present at [:react-native-serial-port-api] AndroidManifest.xml:13:9-36 value=(false).
        Suggestion: add 'tools:replace="android:allowBackup"' to <application> element at AndroidManifest.xml:7:5-117 to override.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processRUIJIA_WIFI_RELEASEDebugManifest'.
> Manifest merger failed : Attribute application@allowBackup value=(true) from AndroidManifest.xml:23:7-33
        is also present at [:react-native-serial-port-api] AndroidManifest.xml:13:9-36 value=(false).
        Suggestion: add 'tools:replace="android:allowBackup"' to <application> element at AndroidManifest.xml:7:5-117 to override.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 55s

    at makeError (/Users/smartop/Desktop/code/smartop-hospital-wifi/node_modules/execa/index.js:174:9)
    at /Users/smartop/Desktop/code/smartop-hospital-wifi/node_modules/execa/index.js:278:16
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async runOnAllDevices (/Users/smartop/Desktop/code/smartop-hospital-wifi/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/runOnAllDevices.js:94:5)
    at async Command.handleAction (/Users/smartop/Desktop/code/smartop-hospital-wifi/node_modules/@react-native-community/cli/build/index.js:186:9)
error Command failed with exit code 1.
```
this is my package.json
```
{
  "name": "smartop-hospital-native-v5",
  "version": "0.0.2",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "jest",
    "lint": "eslint .",
    "build": "react-native run-android --variant=release",
    "RUIJIA_PRE_RELEASE": "cd android && ./gradlew assembleRUIJIA_PRE_RELEASE",
    "RUIJIA_WIFI_RELEASE": "cd android && ./gradlew assembleRUIJIA_WIFI_RELEASE",
    "RUIJIA_PRIVATE": "cd android && ./gradlew assembleRUIJIA_PRIVATE",
    "RUIJIA_PRIVATE_ETHERNET_ONLY": "cd android && ./gradlew assembleRUIJIA_PRIVATE_ETHERNET_ONLY",
    "OFFLINE_VERSION_NEUTRAL": "cd android && ./gradlew assembleOFFLINE_VERSION_NEUTRAL",
    "ZHUHAI_MB_RELEASE": "cd android && ./gradlew assembleZHUHAI_MB_RELEASE",
    "DEMO_BOX_RUIJIA": "cd android && ./gradlew assembleDEMO_BOX_RUIJIA",
    "DEMO_BOX_QUNCHUANG": "cd android && ./gradlew assembleDEMO_BOX_QUNCHUANG"
  },
  "dependencies": {
    "@expo/react-native-action-sheet": "^3.10.0",
    "@react-native-community/async-storage": "^1.11.0",
    "@react-native-community/datetimepicker": "^3.0.3",
    "@react-native-community/masked-view": "^0.1.10",
    "@react-native-community/netinfo": "5.9.10",
    "@react-native-community/picker": "^1.7.1",
    "@react-navigation/bottom-tabs": "^5.5.1",
    "@react-navigation/native": "^5.4.3",
    "@react-navigation/stack": "^5.4.0",
    "async-await-retry": "^1.0.4",
    "axios": "^0.19.2",
    "bluebird": "^3.7.2",
    "chroma-js": "^2.1.0",
    "cross-fetch": "^3.0.5",
    "dotenv-parse-variables": "^1.0.1",
    "lodash": "^4.17.15",
    "mediasoup-client": "^3.6.20",
    "moment": "^2.26.0",
    "promise-retry": "^1.1.1",
    "prop-types": "^15.7.2",
    "protoo-client": "^4.0.4",
    "qs": "^6.9.4",
    "react": "^16.11.0",
    "react-native": "0.62.2",
    "react-native-android-open-settings": "^1.3.0",
    "react-native-android-wifi": "^0.0.41",
    "react-native-animatable": "^1.3.3",
    "react-native-auto-scrolling": "^2.0.0",
    "react-native-awesome-alerts": "^1.3.3",
    "react-native-background-job": "^2.3.1",
    "react-native-background-upload": "5.7.0",
    "react-native-build-config": "^0.3.2",
    "react-native-camera": "git+https://git@github.com/react-native-community/react-native-camera.git",
    "react-native-config": "^1.3.3",
    "react-native-confirmation-code-input": "^1.0.4",
    "react-native-datepicker": "^1.7.2",
    "react-native-device-info": "^5.6.5",
    "react-native-extended-stylesheet": "^0.12.0",
    "react-native-floating-action": "^1.21.0",
    "react-native-fs": "^2.16.6",
    "react-native-gesture-handler": "^1.7.0",
    "react-native-get-random-values": "^1.4.0",
    "react-native-immersive": "^2.0.0",
    "react-native-incall-manager": "^3.3.0",
    "react-native-keep-awake": "^4.0.0",
    "react-native-keyevent": "^0.2.8",
    "react-native-modal": "^11.5.6",
    "react-native-modal-datetime-picker": "^8.7.1",
    "react-native-modalbox": "^2.0.0",
    "react-native-orientation-locker": "^1.1.8",
    "react-native-picker-cascader": "^1.0.0",
    "react-native-picker-select": "^8.0.0",
    "react-native-progress": "^4.1.2",
    "react-native-prompt-android": "^1.1.0",
    "react-native-qrcode-svg": "^6.0.6",
    "react-native-reanimated": "^1.13.0",
    "react-native-restart": "^0.0.17",
    "react-native-root-toast": "^3.2.1",
    "react-native-safe-area-context": "^2.0.3",
    "react-native-screen-brightness": "^2.0.0-alpha",
    "react-native-screens": "^2.10.1",
    "react-native-serial-port-api": "^1.3.0",
    "react-native-simple-table": "^0.0.3",
    "react-native-spinkit": "^1.5.0",
    "react-native-svg": "^12.1.0",
    "react-native-system-setting": "^1.7.5",
    "react-native-table-component": "^1.2.1",
    "react-native-tts": "^3.1.1",
    "react-native-vector-icons": "^6.6.0",
    "react-native-version-number": "^0.3.6",
    "react-native-view-shot": "^3.1.2",
    "react-native-webrtc": "^1.98.0",
    "react-native-webview": "^10.10.2",
    "react-redux": "^7.2.0",
    "redux": "^4.0.5",
    "redux-persist": "^6.0.0",
    "redux-saga": "^1.1.3",
    "rn-test-exception-handler": "^1.3.2",
    "socket.io-client": "2.1.1",
    "uuid": "^8.3.1"
  },
  "devDependencies": {
    "@babel/core": "^7.6.2",
    "@babel/runtime": "^7.6.2",
    "@react-native-community/eslint-config": "^0.0.5",
    "@types/socket.io-client": "1.4.36",
    "@typescript-eslint/eslint-plugin": "^2.8.0",
    "@typescript-eslint/parser": "^2.34.0",
    "babel-jest": "^24.9.0",
    "eslint": "6",
    "eslint-config-airbnb": "^18.1.0",
    "eslint-plugin-import": "^2.20.2",
    "eslint-plugin-jsx-a11y": "^6.2.3",
    "eslint-plugin-react": "^7.20.0",
    "eslint-plugin-react-hooks": "^2.5.0",
    "jest": "^24.9.0",
    "metro-react-native-babel-preset": "^0.58.0",
    "react-test-renderer": "16.11.0",
    "typescript": "^3.9.3"
  },
  "jest": {
    "preset": "react-native"
  }
}
```

**Comment by bastengao** [comment]
I push a new branch `fix-android-manifest` which remove `allowBackup` attribute. I can try to install this branch by yarn again. Please tell me if it works.

**Comment by modifyFun** [comment]
> branch `fix-android-manifest`

how to use branch fix-android-manifest in my project?which command should i exec?

**Comment by bastengao** [comment]
> > branch `fix-android-manifest`
> 
> how to use branch fix-android-manifest in my project?which command should i exec?

@modifyFun May you search `How npm install package with branch` by Google?

**Comment by modifyFun** [comment]
> > > branch `fix-android-manifest`
> > 
> > 
> > how to use branch fix-android-manifest in my project?which command should i exec?
> 
> @modifyFun May you search `How npm install package with branch` by Google?

 I have run successfully by use branch fix-android-manifest in same problem.Thank !

**Comment by bastengao** [comment]
@modifyFun @1596944197 I released v1.3.1 including the fix, you can use this version.
